akka-cluster-metrics/src/main/scala/akka/cluster/metrics/MetricsCollector.scala

```diff
@@ -76,7 +76,11 @@ private[metrics] object MetricsCollector {
         create(collectorCustom).orElse(create(collectorSigar)).orElse(create(collectorJMX))
 
     collector.recover {
-      case e => throw new ConfigurationException(s"Could not create metrics collector: ${e}")
+      case e => {
+        if (useCustom && collector.isFailure)
+          log.debug(s"Failed to create custom metrics collector: ${e}")
+        throw new ConfigurationException(s"Could not create metrics collector: ${e}")
+      }
```

Comment:
Isn't `${e}` in the debug log exactly the `${e}` that would go in the `ConfigurationException`?

Shouldn't it rather go as `cause` for the `ConfigurationException` ?

Reply:
I see, the fallback actually happens on L76 or L78. I think it would be better to add logging there (instead of `orElse` somehow). This wouldn't log an intermediate failure anyway since it has lead to `orElse` the next provider `create` method.

Something like an `def logAndOrElse[U](firstChoice: Try[U], fallback: => Try[U]): Try[U]` could do the trick.